bundles/org.openhab.binding.teslascope/src/main/java/org/openhab/binding/teslascope/internal/TeslascopeHandler.java

```diff
@@ -0,0 +1,376 @@
+                    break;
+                default:
+                    logger.debug("Received command ({}) of wrong type for thing '{}' on channel {}", command,
+                            thing.getUID().getAsString(), channelUID.getId());
+            }
+        } catch (TeslascopeAuthenticationException e) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, e.getMessage());
+        } catch (TeslascopeCommunicationException e) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, e.getMessage());
+        }
+    }
+
+    @Override
+    public void initialize() {
+        config = getConfigAs(TeslascopeConfiguration.class);
+        if (config.apiKey.isBlank()) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
+                    "@text/offline.conf-error.no-api-key");
+            return;
+        }
+
+        updateStatus(ThingStatus.UNKNOWN);
+
+        schedulePoll();
+    }
+
+    @Override
+    public void dispose() {
+        super.dispose();
+        stopPoll();
+    }
+
+    private void schedulePoll() {
+        logger.debug("Scheduling poll every {} s", config.refreshInterval);
+        this.pollFuture = scheduler.scheduleWithFixedDelay(this::pollStatus, 0, config.refreshInterval,
+                TimeUnit.SECONDS);
+    }
+
+    private void stopPoll() {
+        final Future<?> future = pollFuture;
+        if (future != null) {
+            future.cancel(true);
+            pollFuture = null;
+        }
+    }
+
+    private void pollStatus() {
+        String response = "";
+
+        try {
+            response = webTargets.getDetailedInformation(config.publicID, config.apiKey);
+            updateStatus(ThingStatus.ONLINE);
+        } catch (TeslascopeAuthenticationException e) {
+            logger.debug("Unexpected authentication error connecting to Teslascope API", e);
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, e.getMessage());
+            return;
+        } catch (TeslascopeCommunicationException e) {
+            logger.debug("Unexpected error connecting to Teslascope API", e);
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, e.getMessage());
+            return;
+        }
+
+        DetailedInformation detailedInformation = gson.fromJson(response, DetailedInformation.class);
+
+        if (detailedInformation != null) {
+            updateState(TeslascopeBindingConstants.CHANNEL_VIN, new StringType(detailedInformation.vin));
+            updateState(TeslascopeBindingConstants.CHANNEL_VEHICLE_NAME, new StringType(detailedInformation.name));
+            updateState(TeslascopeBindingConstants.CHANNEL_VEHICLE_STATE, new StringType(detailedInformation.state));
+            updateState(TeslascopeBindingConstants.CHANNEL_ODOMETER,
+                    new QuantityType<>(detailedInformation.odometer, ImperialUnits.MILE));
```

Comment:
Just a check; `detailedInformation.odometer` always reports in mile? Or can it also be km dependingw on some user setting in the teslascope profile / account?

Same question applies to other imperial / metric related units.

Reply:
always miles - it's weird as some Tesla API report imperial, and some report metric (ie the temperature ones)